src/Reflection/ReflectionProperty.php

```diff
@@ -118,6 +119,33 @@ private function getVisibilityAsString()
     }
 
     /**
+     * Set the default visibility of this propery. Use the core \ReflectionProperty::IS_* values as parameters, e.g.:
+     *
+     * @param int $newVisibility
+     * @throws \InvalidArgumentException
+     */
+    public function setVisibility($newVisibility)
+    {
+        $this->node->type = ~Class_::MODIFIER_PRIVATE;
```

Comment:
Could you point me at where this is happening? This can be simply reduced into a single bitwise operation:

`$this->node->type = ~Class_::MODIFIER_PRIVATE & ~Class_::MODIFIER_PROTECTED & ~Class_::MODIFIER_PUBLIC`

As it currently stands, your last assignment (line 131) is setting the modifier to "PUBLIC", and that's it.